#!/usr/bin/env python
PACKAGE = "target_tracking_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("flat_vel_kp",    double_t,    0, "Proportional", 1.0,  0, 10)
gen.add("flat_vel_ki",    double_t,    0, "Integral",    .0,   0, 10)
gen.add("flat_vel_deriv_prediction_dt",    double_t,    0, "Derivative prediction time interval",  0.,   0, 1)
gen.add("flat_vel_max_deriv_noise_gain",    double_t,    0, "Max derivative noise gain",  1.0,   0, 10)

gen.add("flat_yaw_kp",    double_t,    0, "Proportional", 0.5,  0, 10)
gen.add("flat_yaw_ki",    double_t,    0, "Integral",    .0,   0, 10)
gen.add("flat_yaw_deriv_prediction_dt",    double_t,    0, "Derivative prediction time interval",  0.7,   0, 1)
gen.add("flat_yaw_max_deriv_noise_gain",    double_t,    0, "Max derivative noise gain",  1.0,   0, 10)

gen.add("flat_pitch_kp",    double_t,    0, "Proportional", 0.5,  0, 10)
gen.add("flat_pitch_ki",    double_t,    0, "Integral",    .0,   0, 10)
gen.add("flat_pitch_deriv_prediction_dt",    double_t,    0, "Derivative prediction time interval",  0.7,   0, 1)
gen.add("flat_pitch_max_deriv_noise_gain",    double_t,    0, "Max derivative noise gain",  1.0,   0, 10)


gen.add("deadzone_abs_vel_error",  double_t, 0,
        "Vel errors in this interval around zero will be ignored",  0.03, 0, 0.5)
gen.add("deadzone_abs_yaw_error",  double_t, 0,
        "Yaw errors in this interval around zero will be ignored",  0.03, 0, 0.5)
gen.add("deadzone_abs_pitch_error",  double_t, 0,
        "Depth errors in this interval around zero will be ignored",  0.05, 0, 0.5)

gen.add("magnify_speed",  double_t, 0, "Forward speed is multiplied by this",  1, 0, 20)
gen.add("sec_before_giving_up",  int_t, 0, "The amount of time after which a DRP observation isn't valid",  1, 0, 10)





exit(gen.generate(PACKAGE, "target_tracking_controller", "DRPControllerParams"))
